Include Help, About, Force Update, Stable, and Dev Commands.

ExtremeFiretop opened the pull request:
Include Help, About, Force Update, Stable, and Dev Commands.

ExtremeFiretop commented on MerlinAU.sh:
```diff
@@ -555,7 +602,7 @@ _CheckForNewScriptUpdates_()
    then
       scriptUpdateNotify="New script update available.
-${REDct}v$SCRIPT_VERSION${NOct} --> ${GRNct}v$DLRepoVersion${NOct}"
+${REDct}v${SCRIPT_VERSION}${NOct} --> ${GRNct}v$DLRepoVersion${NOct}"
```
Identified a little "bug" with the update function here.
Don't think it had any impact, but my editor was detecting this as a variable called "$SCRIPT_VERSION$" instead of "$SCRIPT_VERSION"

Martinski4GitHub replied:
> Identified a little "bug" with the update function here. Don't think it had any impact, but my editor was detecting this as a variable called "$SCRIPT_VERSION$" with the extra dollar sign at the end, instead of "$SCRIPT_VERSION"
>
>
Good catch!! And you're right; the way it was had no effect on the actual functionality because the shell interpreter parser is smart enough to distinguish the 2 variable references within the expression. However, it's always recommended to clearly & unambiguously separate each variable reference when they are next to each other like that, and it's also considered a best-practice coding standard.

ExtremeFiretop replied:
Exactly. I know the shell interpreter is smart enough, this is probably a limitation of my editor, but I figured it's an easy cleanup while I was here.

ExtremeFiretop commented on MerlinAU.sh:
```diff
@@ -587,7 +656,7 @@ _SCRIPTUPDATE_()
       then
           echo ; echo
-          echo -e "${CYANct}Downloading $SCRIPT_NAME ${CYANct}v$DLRepoVersion${NOct}"
+          echo -e "${CYANct}Downloading $SCRIPT_NAME ${CYANct}v${DLRepoVersion}${NOct}"
```
Same with these: $DLRepoVersion$ being detected as a different variable than $DLRepoVersion

Martinski4GitHub replied:
> Same with these: `$DLRepoVersion$` being detected as a different variable than `$DLRepoVersion`
>
Another good catch!! It is best to follow the best-practice coding standards.

Martinski4GitHub commented:
> Include Help, About, Force Update, Stable, and Dev Commands.
>
I'll review this PR tomorrow. My wife & I started binge-watching a short series on Amazon Prime Video ("Outer Range").
Right now we're on a "bathroom break" and I saw your messages.

We'll talk tomorrow.

ExtremeFiretop commented:
> > Include Help, About, Force Update, Stable, and Dev Commands.
> 
> I'll review this PR tomorrow. My wife & I started binge-watching a short series on Amazon Prime Video ("Outer Range"). Right now we're on a "bathroom break" and I saw your messages.
> 
> We'll talk tomorrow.

Enjoy the show!

ExtremeFiretop commented:
@Martinski4GitHub 

I updated the version.txt and the version number in the dev branch early, so this can be tested for you :)
While testing, I identified a little mistake, and fixed it in commit: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/260/commits/31edb31e17b96f1f8746c403e4319c9e649d05b0

Martinski4GitHub commented on MerlinAU.sh:
```diff
@@ -233,6 +233,67 @@ _DoExit_()
+  $SCRIPT_NAME uninstall          uninstalls script
+  $SCRIPT_NAME addCronJob         add the MerlinAU cron job
+  $SCRIPT_NAME postRebootRun      add the post-update reboot cron job
+  $SCRIPT_NAME postUpdateEmail    Send post-update email
```
The last 3 call parameters (**addCronJob**, **postRebootRun**, **postUpdateEmail**) are intended **only** for internal use to handle specific events and are not meant to be exposed to users for CLI calls. Those lines should be removed from the "`_ShowHelp_()`"  function.

ExtremeFiretop replied:
No worries! I can remove them from the help function :)

I was just including them all while I was at it, I wasn't sure if we should or shouldn't include the original commands so good to know we can keep those hidden.

ExtremeFiretop replied:
Done here: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/260/commits/65eb51d4f4e633c47ba101ee2292ebf1fbb27f4f

Martinski4GitHub commented:
My apologies for the delayed response. My "Higher Power :>)"  summoned & asked me to go to the store and buy some groceries and other stuff for tomorrow's family gathering (we're having a 4th of July barbecue) and then helped in the kitchen.

Anyway, I've tested the PR &  overall it's working well. However, there are a few places that need some fine-tuning & refactoring. I'll submit a separate PR with my proposed changes.

Martinski4GitHub reviewed (approved):
Working well so it's approved.
I'll submit a separate PR to apply some refactoring & fine-tuning.

ExtremeFiretop commented:
> My apologies for the delayed response. My "Higher Power :>)" summoned & asked me to go to the store and buy some groceries and other stuff for tomorrow's family gathering (we're having a 4th of July barbecue) and then helped in the kitchen.
> 
> Anyway, I've tested the PR & overall it's working well. However, there are a few places that need some fine-tuning & refactoring. I'll submit a separate PR with my proposed changes.

Ah yes! The celebrations are incoming!
My Canada day was fairly relaxed, spent some time with the family as well, had a BBQ too!

Nice fireworks at the end of the evening, overall a good day :) I'm sure you looking forwards to well earned time off!

Anyways, no worries it's not a rush, thanks for testing the PR! Feel free to submit some adjustments. Is there anything non functional I missed?

ExtremeFiretop commented:
> Working well so it's approved. I'll submit a separate PR to apply some refactoring & fine-tuning.


That answers my question :)

ExtremeFiretop commented:
Synced with Gnuton here: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/186/commits/a3dd4551e6c9f3f9731d91396b0000f6e539bfc7
